File: ios/NetrakshAI/NetrakshConfigModule.swift
import Foundation
import React

@objc(NetrakshConfig)
class NetrakshConfig: NSObject {
  @objc
  static func requiresMainQueueSetup() -> Bool {
    false
  }

  @objc
  func constantsToExport() -> [AnyHashable: Any]! {
    ["runtimeConfig": runtimeConfig()]
  }

  private func runtimeConfig() -> [String: Any] {
    var config: [String: Any] = [:]

    putString(&config, "apiBaseUrl", plistString("NETRAKSH_API_BASE_URL"))
    putString(&config, "apiTenantId", plistString("NETRAKSH_API_TENANT_ID"))
    putString(&config, "apiSiteId", plistString("NETRAKSH_API_SITE_ID"))
    putInt(&config, "apiTimeoutMs", plistInt("NETRAKSH_API_TIMEOUT_MS"))
    putInt(
      &config,
      "authLogSyncIntervalMs",
      plistInt("NETRAKSH_AUTH_LOG_SYNC_INTERVAL_MS")
    )
    putBool(
      &config,
      "databaseEncryptionEnabled",
      plistBool("NETRAKSH_DATABASE_ENCRYPTION_ENABLED")
    )
    putString(
      &config,
      "databaseLocation",
      plistString("NETRAKSH_DATABASE_LOCATION")
    )
    putString(&config, "databaseName", plistString("NETRAKSH_DATABASE_NAME"))
    putString(
      &config,
      "databaseProvider",
      plistString("NETRAKSH_DATABASE_PROVIDER")
    )
    putInt(
      &config,
      "databaseSchemaVersion",
      plistInt("NETRAKSH_DATABASE_SCHEMA_VERSION")
    )
    putBool(&config, "demoMode", plistBool("NETRAKSH_DEMO_MODE"))

    return config
  }

  private func plistString(_ key: String) -> String? {
    guard let value = Bundle.main.object(forInfoDictionaryKey: key) as? String else {
      return nil
    }

    let trimmed = value.trimmingCharacters(in: .whitespacesAndNewlines)
    if trimmed.isEmpty || (trimmed.hasPrefix("$(") && trimmed.hasSuffix(")")) {
      return nil
    }

    return trimmed
  }

  private func plistInt(_ key: String) -> Int? {
    guard let value = plistString(key) else {
      return nil
    }

    return Int(value)
  }

  private func plistBool(_ key: String) -> Bool? {
    guard let value = plistString(key)?.lowercased() else {
      return nil
    }

    if ["true", "1", "yes"].contains(value) {
      return true
    }

    if ["false", "0", "no"].contains(value) {
      return false
    }

    return nil
  }

  private func putString(_ config: inout [String: Any], _ key: String, _ value: String?) {
    if let value {
      config[key] = value
    }
  }

  private func putInt(_ config: inout [String: Any], _ key: String, _ value: Int?) {
    if let value {
      config[key] = value
    }
  }

  private func putBool(_ config: inout [String: Any], _ key: String, _ value: Bool?) {
    if let value {
      config[key] = value
    }
  }
}

File: android/app/src/main/java/com/netrakshai/config/NetrakshConfigModule.java
package com.netrakshai.config;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.netrakshai.BuildConfig;
import java.util.HashMap;
import java.util.Map;

public class NetrakshConfigModule extends ReactContextBaseJavaModule {
  public static final String NAME = "NetrakshConfig";

  public NetrakshConfigModule(ReactApplicationContext reactContext) {
    super(reactContext);
  }

  @Override
  public String getName() {
    return NAME;
  }

  @Override
  public Map<String, Object> getConstants() {
    Map<String, Object> runtimeConfig = new HashMap<>();

    putString(runtimeConfig, "apiBaseUrl", BuildConfig.NETRAKSH_API_BASE_URL);
    putString(runtimeConfig, "apiTenantId", BuildConfig.NETRAKSH_API_TENANT_ID);
    putString(runtimeConfig, "apiSiteId", BuildConfig.NETRAKSH_API_SITE_ID);
    putInteger(runtimeConfig, "apiTimeoutMs", BuildConfig.NETRAKSH_API_TIMEOUT_MS);
    putInteger(
        runtimeConfig,
        "authLogSyncIntervalMs",
        BuildConfig.NETRAKSH_AUTH_LOG_SYNC_INTERVAL_MS);
    putBoolean(
        runtimeConfig,
        "databaseEncryptionEnabled",
        BuildConfig.NETRAKSH_DATABASE_ENCRYPTION_ENABLED);
    putString(
        runtimeConfig,
        "databaseLocation",
        BuildConfig.NETRAKSH_DATABASE_LOCATION);
    putString(runtimeConfig, "databaseName", BuildConfig.NETRAKSH_DATABASE_NAME);
    putString(
        runtimeConfig,
        "databaseProvider",
        BuildConfig.NETRAKSH_DATABASE_PROVIDER);
    putInteger(
        runtimeConfig,
        "databaseSchemaVersion",
        BuildConfig.NETRAKSH_DATABASE_SCHEMA_VERSION);
    putBoolean(runtimeConfig, "demoMode", BuildConfig.NETRAKSH_DEMO_MODE);

    Map<String, Object> constants = new HashMap<>();
    constants.put("runtimeConfig", runtimeConfig);
    return constants;
  }

  private static void putString(
      Map<String, Object> config,
      String key,
      String value) {
    if (value != null && !value.trim().isEmpty()) {
      config.put(key, value.trim());
    }
  }

  private static void putInteger(
      Map<String, Object> config,
      String key,
      String value) {
    if (value == null || value.trim().isEmpty()) {
      return;
    }

    try {
      config.put(key, Integer.parseInt(value.trim()));
    } catch (NumberFormatException ignored) {
      // Invalid optional runtime values are ignored so JS defaults stay active.
    }
  }

  private static void putBoolean(
      Map<String, Object> config,
      String key,
      String value) {
    if (value == null || value.trim().isEmpty()) {
      return;
    }

    String normalized = value.trim().toLowerCase();
    if ("true".equals(normalized) || "1".equals(normalized) || "yes".equals(normalized)) {
      config.put(key, true);
    } else if (
        "false".equals(normalized) ||
        "0".equals(normalized) ||
        "no".equals(normalized)) {
      config.put(key, false);
    }
  }
}
